DepositApi.BLL/Services/DepositService.cs

```diff
@@ -9,21 +12,61 @@ namespace DepositApi.BLL.Services
 {
     public class DepositService : IDepositService
     {
-        public DepositService() { }
+        private readonly IRepository<Deposit> depositRepository;
+        private readonly IRepository<DepositCalc> depositCalcRepository;
+        private readonly IMapper mapper;
 
-        public List<DepositCalcDTO> PersentCalculationAsync(DepositDTO depositDTO)
+        public DepositService(IRepository<Deposit> depositRepository, IRepository<DepositCalc> depositCalcRepository, IMapper mapper)
+        {
+            this.depositRepository = depositRepository;
+            this.depositCalcRepository = depositCalcRepository;
+            this.mapper = mapper;
+        }
+
+        public async Task<List<DepositCalcDTO>> PersentCalculationAsync(DepositDTO depositDTO)
         {
             var result = new List<DepositCalcDTO>();
-            for(int i = 1; i <= depositDTO.Term; i ++)
+            depositDTO.Date = DateTime.UtcNow.Date;
+            var deposit = this.mapper.Map<Deposit>(depositDTO);
+            await this.depositRepository.CreateAsync(deposit);
+            for (int i = 1; i <= depositDTO.Term; i ++)
             {
                 result.Add(new DepositCalcDTO
                 {
                     Month = i,
                     PercentAdded = Math.Round(depositDTO.Amount * (depositDTO.Percent / 12 / 100), 2, MidpointRounding.AwayFromZero),
                     TotalAmount = Math.Round(depositDTO.Amount * (1 + depositDTO.Percent / 12 / 100 * i), 2, MidpointRounding.AwayFromZero),
+                    DepositId = deposit.Id
                 });
             }
+            var depositCalcs = mapper.Map<List<DepositCalc>>(result);
+            await this.depositCalcRepository.CreateRangeAsync(depositCalcs);
             return result;
         }
+
+        public async Task<List<DepositDTO>> GetDepositsAsync(int startIndex = 0, int count = 20)
+        {
+            var deposits = await this.depositRepository.FindRangeAsync(d => true, startIndex, count);
+            var depositsDTO = this.mapper.Map<List<DepositDTO>>(deposits);
+            return depositsDTO;
+        }
+
+        public async Task<List<DepositCalcDTO>> GetDepositCalcsAsync(int depositId)
+        {
+            var depositCalcs = await this.depositCalcRepository.FindRangeAsync(d => d.DepositId == depositId, 0, 100);
+            var depositCalcDTOs = this.mapper.Map<List<DepositCalcDTO>>(depositCalcs);
+            return depositCalcDTOs;
+        }
+
+        public async Task<byte[]> GetDepositCalcCSVAsync(int depositId)
+        {
+            string result = string.Empty;
+            var depositCalcs = await this.depositCalcRepository.FindRangeAsync(d => d.DepositId == depositId, 0, 100);
+            foreach (DepositCalc depositCalc in depositCalcs)
+            {
+                result += $"{depositCalc.Month},{depositCalc.PercentAdded},{depositCalc.TotalAmount}\n";
+            }
+            return Encoding.ASCII.GetBytes(result);
```

Comment:
Regarding to the best practise of the code clean , we should add an empty line before the returning value line.